From 0fd8879170eabd1cd49968781ccc87188e2a2fc4 Mon Sep 17 00:00:00 2001
From: Jiani Liu <quic_jianil@quicinc.com>
Date: Fri, 7 Apr 2023 10:49:55 +0800
Subject: [PATCH] ath10k: Register reg_notifier when regdomain is forced

When CONFIG_QCA_ATH_USER_REGD is defined make sure to register
reg_notifier to configure channels and reg domain information
to fw when user changes country code.

Change-Id: I2ecb54583c251c4cbab3b15d426314e383176818
Signed-off-by: Rajkumar Manoharan <rmanohar@codeaurora.org>
Signed-off-by: Jiani Liu <quic_jianil@quicinc.com>

diff --git a/drivers/net/wireless/ath/ath10k/mac.c b/drivers/net/wireless/ath/ath10k/mac.c
index 4b34fe0..aab65bc 100644
--- a/drivers/net/wireless/ath/ath10k/mac.c
+++ b/drivers/net/wireless/ath/ath10k/mac.c
@@ -10148,6 +10148,9 @@ int ath10k_mac_register(struct ath10k *ar)
 
 	ret = ath_regd_init(&ar->ath_common.regulatory, ar->hw->wiphy,
 			    ath10k_reg_notifier);
+#ifdef CPTCFG_ATH_USER_REGD
+	ar->hw->wiphy->reg_notifier = ath10k_reg_notifier;
+#endif
 	if (ret) {
 		ath10k_err(ar, "failed to initialise regulatory: %i\n", ret);
 		goto err_dfs_detector_exit;
-- 
2.7.4

